share/ansible_navigator/utils/catalog_collections.py

```diff
@@ -97,6 +99,98 @@ def _catalog_plugins(self, collection: Dict) -> None:
                     collection,
                 )
 
+    def _catalog_roles(self, collection: Dict) -> None:
+        """Catalog the roles within a collection.
+
+        :param collection: Details describing the collection
+        """
+        # pylint: disable=too-many-locals
+        # pylint: disable=too-many-statements
+
+        collection_name: str = collection["known_as"]
+        collection["roles"] = []
+        roles_directory = Path(collection["path"], "roles")
+        if not roles_directory.is_dir():
+            return
+
+        for role_directory in roles_directory.iterdir():
+            role: Dict[str, Union[str, Dict[str, Any]]] = {
```

Comment:
The shape of the defaults file in a role could be a Dict[str, .........   of unknown depth. Yes it would need to be a basc type, but I don't think  we can expressive depth yet...

This is terrible to look at:

```
_JSONType_0 = t.Union[str, int, float, bool, None, t.Dict[str, t.Any], t.List[t.Any]]
_JSONType_1 = t.Union[str, int, float, bool, None, t.Dict[str, _JSONType_0], t.List[_JSONType_0]]
_JSONType_2 = t.Union[str, int, float, bool, None, t.Dict[str, _JSONType_1], t.List[_JSONType_1]]
_JSONType_3 = t.Union[str, int, float, bool, None, t.Dict[str, _JSONType_2], t.List[_JSONType_2]]
JSONType =  t.Union[str, int, float, bool, None, t.Dict[str, _JSONType_3], t.List[_JSONType_3]]
```
https://github.com/python/typing/issues/182